Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp

```diff
@@ -520,8 +521,22 @@ namespace AZ::SceneAPI::Behaviors
 
         // AssetImportRequest
         Events::ProcessingResult UpdateManifest(Containers::Scene& scene, ManifestAction action, RequestingApplication requester) override;
+        Events::ProcessingResult PrepareForAssetLoading(Containers::Scene& scene, RequestingApplication requester) override;
     };
 
+    Events::ProcessingResult PrefabGroupBehavior::ExportEventHandler::PrepareForAssetLoading(
+        [[maybe_unused]] Containers::Scene& scene,
+        RequestingApplication requester)
+    {
+        using namespace AzToolsFramework;
+        if (requester == RequestingApplication::AssetProcessor)
+        {
+            EntityUtilityBus::Broadcast(&EntityUtilityBus::Events::ResetEntityContext);
+            AZ::Interface<AzToolsFramework::Prefab::PrefabSystemComponentInterface>::Get()->RemoveAllTemplates();
```

Comment:
@jackalbe , does this mean that during the loading of a particular proc prefab, every prefab could get wiped out from memory? This could be expensive because if other assets depend on some of the wiped out prefabs, they'll again need to be loaded into templates from disk. If the intention is to wipe certain prefab templates, RemoveTemplate() should be used for them instead of RemoveAllTemplates(). For fixing the crash, this PR should be good enough but this logic should be revisited to avoid potential slowdowns of AP in the future. Feel free to reach out to me if you want to discuss a possible better solution here.